client/signup/steps/pressable-store/index.jsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import EmailValidator from 'email-validator';
+
+/**
+ * Internal dependencies
+ */
+import StepWrapper from 'signup/step-wrapper';
+import Gridicon from 'components/gridicon';
+
+import LoggedOutForm from 'components/logged-out-form';
+import LoggedOutFormFooter from 'components/logged-out-form/footer';
+import LoggedOutFormLinks from 'components/logged-out-form/links';
+import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
+import FormTextInput from 'components/forms/form-text-input';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormButton from 'components/forms/form-button';
+import FormLabel from 'components/forms/form-label';
+import FormSectionHeading from 'components/forms/form-section-heading';
+
+import HeroImage from './hero-image';
+
+export default React.createClass( {
+	displayName: 'PressableStoreStep',
+
+	propTypes: {
+		stepName: PropTypes.string.isRequired,
+		goToNextStep: PropTypes.func.isRequired,
+		signupDependencies: PropTypes.object.isRequired,
+	},
+
+	getInitialState() {
+		return {
+			email: '',
+			valid: false,
+			error: null,
+		};
+	},
+
+	componentDidMount() {
+		if ( this._input ) {
+			this._input.focus();
+		}
+	},
+
+	onEmailChange( event ) {
+		const email = event.target.value;
+		const valid = EmailValidator.validate( email );
+		this.setState( {
+			email,
+			valid,
+		} );
+		if ( this.state.error && valid ) {
+			this.setState( {
+				error: null,
+			} );
+		}
+	},
+
+	onSubmit( event ) {
+		event.preventDefault();
+
+		if ( ! this.state.valid ) {
+			this.setState( {
+				error: this.translate( 'Please provide a valid email address.' ),
+			} );
+			return;
+		}
+
+		window.location.href = `https://my.pressable.com/signup/five-sites?email=${ encodeURIComponent( this.state.email ) }&wp-ecommerce=true`;
+	},
+
+	onEmailInputRef( input ) {
+		this._input = input;
+	},
+
+	renderStoreForm() {
+		return (
+			<div>
+				<LoggedOutForm className="pressable-store__form">
+					<HeroImage />
+
+					<FormSectionHeading className="pressable-store__heading">{ this.translate( 'Get your store for as low as $25 / month' ) }</FormSectionHeading>
+					<p className="pressable-store__copy">{ this.translate( 'We\'ve partnered with Pressable, a top-notch WordPress hosting provider, and WooCommerce, the go-to eCommerce solution for WordPress, to make setting up your store a snap.' ) }</p>
+
+					<LoggedOutFormFooter>
+						<FormLabel for="email">{ this.translate( 'Start by entering your email address:' ) }</FormLabel>
+						<div className="pressable-store__form-fields">
+							<FormTextInput ref={ this.onEmailInputRef } isError={ this.state.error } isValid={ this.state.valid } onChange={ this.onEmailChange } className="pressable-store__form-email is-spaced" type="email" placeholder="Email Address" name="email" />
```

Comment:
Not familiar with the domain, but is this passing of `ref` doing anything? From the looks of `FormTextInput`, it [doesn't look like it](https://github.com/Automattic/wp-calypso/blob/d8113bc8d0c409a675618395d434b12906277fb2/client/components/forms/form-text-input/index.jsx).

By extension: is `onEmailInputRef` being called? I realize that I could just test myself, but I'll ask anyway. :)